reference/TubePlayer/TubePlayer/Business/YoutubeServiceMock.cs

```diff
@@ -15,6 +15,11 @@ public YoutubeServiceMock(ISerializer serializer)
         _channels = channelsData.Items!.ToDictionary(channel => channel.Id!, StringComparer.OrdinalIgnoreCase);
     }
 
+    public Task<string?> GetVideoSourceUrl(string videoId, CancellationToken ct)
+    {
+        return Task.FromResult<string?>(null);
```

Comment:
```suggestion
        return Task.FromResult<string?>(default);
```

Reply:
Alternatively, we should return a fix url so the video details page will actually play a video. Perhaps use one of the buck bunny sample video urls

Reply:
@nickrandolph  was investigating, and unfortunately, we cannot set a constant URL here since this has metadata information that changes with every request, and also, most importantly, this URL has an expiration time attached to it.

Sample Url: 

```
https://rr2---sn-8qu-t0aee.googlevideo.com/videoplayback?expire=1700524761&ei=eZ5bZd_gHYee_9EPyKObmAE&itag=22&source=youtube&requiressl=yes&mh=V-&mm=31%2C29&mn=sn-8qu-t0aee%2Csn-t0a7lnee&ms=au%2Crdu&mv=m&mvi=2&pl=18&initcwndbps=1572500&vprv=1&mime=video%2Fmp4&cnr=14&ratebypass=yes&dur=96.084&lmt=1698927054722942&mt=1700502803&fvip=5&fexp=24007246&beids=24350018&c=ANDROID_TESTSUITE&txp=5318224&sparams=expire%2Cei%2Cip%2Cid%2Citag%2Csource%2Crequiressl%2Cvprv%2Cmime%2Ccnr%2Cratebypass%2Cdur%2Clmt&sig=ANLwegAwRAIgL_wIJ3YDWa56owUQkrstK-4BCfXorYEwjarUZMjL8ugCIG1u9s1uI9KPcLP_5VmDzL0pBWrwxwwgcLLddSXNGubE&lsparams=mh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpl%2Cinitcwndbps&lsig=AM8Gb2swRAIgCBwqBpbN1_Wg-pY_0Ilgma8FqswrKsFVgDbJPQRQTucCICe_3iHix589bpquSgFTXlgmsdFwUXmN5ZlmhZC9Gxac
```